libtransmission/peer-mgr.cc

```diff
@@ -1720,22 +1720,17 @@ void updateInterest(tr_swarm* swarm)
         return;
     }
 
-    if (auto const& peers = swarm->peers; !std::empty(peers))
+    // build a bitfield of interesting pieces...
+    std::vector<bool>::size_type const n = tor->piece_count();
+    auto piece_is_interesting = std::vector<bool>(n);
+    for (size_t i = 0; i < n; ++i)
```

Comment:
I think you might be misreading the previous version as `for(peer : peers)` instead of `if (peers = foo(); !std::empty(peers))` ?

It was already outside the loop in the old version & had the bonus of not needing to build `piece_is_interesting` unless there is at least one connected peer.

Reply:
Oops, yes you are right.